src/content/platform-integration/platform-channels.md

```diff
@@ -1378,45 +1357,127 @@ new Handler(Looper.getMainLooper()).post(new Runnable() {
 });
 ```
 
-### Jumping to the main thread in iOS
+{% endtab %}
+
+{% endtabs %}
+
+### Jump to the main thread (iOS) {: #jumping-to-the-main-thread-in-ios }
 
 To comply with channel's main thread requirement,
 you might need to jump from a background thread to
 iOS's main thread to execute a channel method.
 You can accomplish this in iOS by executing a
 [block][] on the main [dispatch queue][]:
 
-In Objective-C:
+{% tabs "lang-tabs" %}
+
+{% tab "Objective-C" %}
 
 ```objc
 dispatch_async(dispatch_get_main_queue(), ^{
   // Call the desired channel message here.
 });
 ```
 
-In Swift:
+{% endtab %}
+
+{% tab "Swift" %}
 
 ```swift
 DispatchQueue.main.async {
   // Call the desired channel message here.
 }
 ```
 
+{% endtab %}
+
+{% endtabs %}
+
+## Supplementals
+
+### Common channels and codecs {:#codec2}
+
+The following is a list of some common platform channel APIs
+that you can use to write platform-specific code:
+
+* [`MethodChannel`][] for Flutter: A named channel that you
+  can use to communicate with platform plugins using
+  asynchronous method calls. By default this channel uses
+  the [`StandardMessageCodec`][] codec.
+  This channel is not type safe, which means calling and
+  receiving messages depends on the host and client
+  declaring the same arguments and data types in order for
+  messages to work.
+
+* [`BasicMessageChannel`][] for Flutter: A named channel
+  that supports basic, asynchronous message passing, using a
+  supported message codec. Not type safe.
+
+* [Engine Embedder APIs][] for Platforms: These
+  platform-specific APIs contain platform-specific
+  channel APIs.
+
+You can create your own codec or use an existing one. The
+following is a list of some existing codecs that you can use
+with platform-specific code:
+
+* [`StandardMessageCodec`][]: A commonly used message codec
+  that encodes and decodes a wide range of data types into
+  a platform-agnostic binary format for transmission across
+  platform channels. The serialization and deserialization
+  of values to and from messages happens automatically when
+  you send and receive values. For a list of supported
+  data types, see [Platform channel data types support](#codec).
+
+* [`BinaryCodec`][]: A message codec that passes raw binary
+  data between the Dart side of your Flutter app and the
+  native platform side. It does not perform any higher-level
+  encoding or decoding of data structures.
+
+* [`StringCodec`][]: A message codec that encodes and
+  decodes strings, using UTF-8 encoding.
+
+* [`JSONMessageCodec`][]: A message codec that encodes and
+  decodes JSON-formatted data, using UTF-8 encoding.
+
+* [`FirestoreMessageCodec`][]: A message codec that handles
+  the exchange of messages sent across the platform channel
+  between your Flutter app and the native
+  Firebase Firestore SDKs (on Android and iOS).
+
+[MessageCodec]: https://api.flutter.dev/flutter/services/MessageCodec-class.html
```

Comment:
I notice that you don't use the macro for the API site. Why?
```suggestion
[MessageCodec]: {{site.api}}/flutter/services/MessageCodec-class.html
```

Reply:
I missed that. Sorry @sfshaza2!